python/lsst/daf/butler/registry/_registry.py

```diff
@@ -870,7 +870,7 @@ def getDataset(self, id: DatasetId) -> Optional[DatasetRef]:
 
         Parameters
         ----------
-        id : `DatasetId`
+        id : `uuid.UUID`
```

Comment:
Formally this really should be a `lsst.daf.butler.DatasetId` which is a type alias for UUID. It seems that Sphinx handles the aliasing just fine in the formal type annotations but gets confused in the numpydoc and doesn't link it properly.

@TallJimbo are we formally deprecating `DatasetId` or do we want to continue to use it as a public-facing alias?

Reply:
FTR, https://pipelines.lsst.io/v/d_2023_06_05/modules/lsst.daf.butler/index.html#classes includes `DatasetId` in its class list, but doesn't have a valid anchor for it (Sphinx warns about this as well; I'd assumed it couldn't be fixed short of removing all references to `DatasetId`). If the documentation of `DatasetId` could be fixed so that it's linkable, then I'd be fine with leaving these references as-is.

Reply:
Ok. Leave it as it was originally and we can try to fix DatasetId as @andy-slac describes above. There is a doc string attached to `DatasetId` in `ref.py` but clearly that doesn't do anything useful.